OpenQA/Isotovideo/Utils.pm

```diff
@@ -158,6 +157,8 @@ sub handle_generated_assets ($command_handler, $clean_shutdown) {
         bmwqemu::diag "Requested to force the publication of '$name'";
         push @toextract, _store_asset($i, $name, 'assets_public');
     }
+    # Nothing to do here if we don't have any assets to extract
+    return unless @toextract;
```

Comment:
I think we aren't using `return` without a value.
You should return `$return_code`, however, that is already done after the loop (which isn't executed if the array is empty anyway), so I don't see how that would change anything.

Reply:
We are in line 131 ;-) But that wasn't the point.
My intention was to skip `$bmwqemu::backend->extract_assets($asset)` which calls `do_extract_assets` even though for example `PUBLISH_HDD_1` was defined. That's what https://openqa.suse.de/tests/7917723 / mru-install-minimal-with-addons@s390x-kvm-sle12 does.

Reply:
Maybe I'm not tracing the code path correctly. Maybe line 131 is not what breaks the abve test. I don't have a unit test for this yet.

Reply:
Well, currently the code would only enter the loop and call `extract_assets` if there are elements in `@toextract`.
Returning before the loop doesn't make a difference in this regard.
Simplified code:
```
return unless @array;
for my $element (@array) {
    do_something($element);
}
return;
```
The first return statement is useless here if `@array` is empty.

The fact that `extract_assets` was called in our case, shows that `@toextract` was not empty.